sqlmesh/core/plan/evaluator.py

```diff
@@ -218,30 +218,36 @@ def _push(
             plan: The plan to source snapshots from.
             deployability_index: Indicates which snapshots are deployable in the context of this creation.
         """
-        snapshots_to_create = [
+        snapshots_to_maybe_create = [
             s
             for s in snapshots.values()
             if s.is_model and not s.is_symbolic and plan.is_selected_for_backfill(s.name)
         ]
-        snapshots_to_create_count = len(snapshots_to_create)
 
-        if snapshots_to_create_count > 0:
+        snapshots_to_create = self.snapshot_evaluator.get_snapshots_to_create(
+            snapshots_to_maybe_create,
+            deployability_index=deployability_index,
+        )
+
+        if snapshots_to_create["snapshots_to_create"]:
```

Comment:
I'm not familiar with `TypedDict` but do you still have to access its members using strings? If so it seems like an object would be more appropriate

Reply:
Ah, good point - I converted it to a pydantic model now. Does that resolve the issue?